AirCasting/CreateSessionViews/ConnectingABViewModel.swift

```diff
@@ -0,0 +1,29 @@
+// Created by Lunar on 21/07/2021.
+//
+
+import CoreBluetooth
+import Foundation
+
+protocol ConnectingABProtocol {
+    var bluetoothManager: BluetoothManager { get }
+    func performConnectingWithin10Second(peripheral: CBPeripheral, completion: @escaping () -> Void)
+}
+
+class ConnectingABViewModel: ConnectingABProtocol {
+    var bluetoothManager: BluetoothManager
+    
+    init(bluetoothManager: BluetoothManager) {
+        self.bluetoothManager = bluetoothManager
+    }
+    
+    func performConnectingWithin10Second(peripheral: CBPeripheral, completion: @escaping () -> Void) {
+        bluetoothManager.centralManager.connect(peripheral, options: nil)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 10) { [self] in
```

Comment:
All this is business and service logic that should be separated from views (see: giant comment ;) )